front/pages/w/[wId]/u/chat/[cId]/index.tsx

```diff
@@ -111,12 +116,31 @@ export const getServerSideProps: GetServerSideProps<{
     };
   }
 
+  let stickyDataSourceSelection: { [key: string]: boolean } = {};
+
+  if (user) {
+    const stickyDataSourceSelectionRaw = await getUserMetadata(
+      user,
+      "chat-data-sources-selection"
+    );
+    if (stickyDataSourceSelectionRaw) {
+      try {
+        stickyDataSourceSelection = JSON.parse(
+          stickyDataSourceSelectionRaw.value
+        );
+      } catch (e) {
+        console.error("Error parsing sticky data source selection", e);
+      }
+    }
+  }
+
   const dataSources: DataSource[] = dsRes.value.map((ds) => {
     return {
       name: ds.name,
       description: ds.description,
       provider: ds.connectorProvider || "none",
-      selected: ds.assistantDefaultSelected,
+      selected:
+        stickyDataSourceSelection[ds.name] ?? ds.assistantDefaultSelected,
```

Comment:
used `??` here, so it means if a user explicitly turned off a DS, it will remain turned off even if it's default-on at the workspace level

Reply:
SGTM